distrobuilder/main.go

```diff
@@ -755,27 +690,29 @@ if is_lxc_container; then
 		fix_systemd_mask vconsole-setup-kludge@tty1.service
 	fi
 
-	# Workarounds for cloud containers
-	if { [ "${ID}" = "fedora" ] || [ "${ID}" = "rhel" ]; } && [ "${cloudinit_exists}" -eq 1 ]; then
-		fix_nm_force_up eth0
-	fi
-
 	# Workarounds for privileged containers.
 	if ! grep -q 4294967295 /proc/self/uid_map && { [ "${ID}" = "altlinux" ] || [ "${ID}" = "arch" ] || [ "${ID}" = "fedora" ]; }; then
 		fix_ro_paths systemd-networkd.service
 		fix_ro_paths systemd-resolved.service
 	fi
-
-	# Workarounds for NetworkManager in containers
-	if [ "${nm_exists}" -eq 1 ]; then
-		if [ "${ID}" = "ol" ] || [ "${ID}" = "centos" ]; then
-			fix_nm_force_up eth0
-		fi
-
-		fix_nm_link_state eth0
-	fi
 fi
 `
 	os.MkdirAll("/etc/systemd/system-generators", 0755)
 	ioutil.WriteFile("/etc/systemd/system-generators/lxc", []byte(content), 0755)
 }
+
+// addVethUdevRule creates an udev rule that makes eth0 work with NetworkManager
+func addVethUdevRule() {
+	content := `
+# This file was created by distrobuilder.
+#
+# Its purpose is to convince NetworkManager to treat the eth0 veth
+# interface like a regular Ethernet. NetworkManager ordinarily doesn't
+# like to manage the veth interfaces, becaue they are typically configured
+# by container management tooling for specialized purposes.
+
+ACTION=="add|change|move", ENV{ID_NET_DRIVER}=="veth", ENV{INTERFACE}=="eth[0-9]*", ENV{NM_UNMANAGED}="0"
+`
+	os.MkdirAll("/etc/udev/rules.d", 0755)
```

Comment:
@lkundrak Could you please add a check for `/etc/udev`, and only write the file if it's present? I guess that directory should be a good indicator whether `udev` is available.

Reply:
@lkundrak I'll merge the PR once the check is in place.